setup.py

```diff
@@ -28,6 +27,9 @@
     url='https://github.com/aws/chalice',
     packages=find_packages(exclude=['tests']),
     install_requires=install_requires,
+    extras_require={
+        'watchdog': ['watchdog==0.8.3'],
```

Comment:
I prefer this named something else more generic so that if we end up replacing watchdog with a more current library, you don't have to change the install command.